weave-js/src/core/weave.ts

```diff
@@ -100,6 +102,18 @@ export class Weave implements WeaveInterface {
     return this.languageBinding.printGraph(node, indent);
   }
 
+  isExpLogicallyEqual(node1: NodeOrVoidNode, node2: NodeOrVoidNode): boolean {
```

Comment:
would you mind adding a comment explaining the purpose of this function, historical context, and the special condition?